src/parquet/column/reader.cc

```diff
@@ -124,12 +134,16 @@ bool TypedColumnReader<TYPE>::ReadNewPage() {
       // first page with this encoding.
       Encoding::type encoding = page->encoding();
 
-      if (IsDictionaryIndexEncoding(encoding)) {
+      if (IsRleDictionaryIndexEncoding(encoding)) {
```

Comment:
this could just be `IsDictionaryEncoding` (or the original name is fine too) since either PLAIN_DICTIONARY or RLE_DICTIONARY is valid